### ISSUE TITLE
sphinx-build dies under 1.7.0

### ISSUE THREAD

**Issue by bnmnetp** [sphinx-doc/sphinx#4615]
Subject: upgrading to 1.7 causes build to die, reverting to 1.6.3 restores function

### Problem
1.7.0 build dies, same command works on 1.6.3

#### Procedure to reproduce the problem
```
sphinx-build  -b html -d ./build/thinkcspy/doctrees -c . -a -E -Acourse_id=thinkcspy -Alogin_required=false -Aappname=runestone -Aloglevel=10 -Acourse_url=http://127.0.0.1:8000 -Ause_services=true -Apython3=true -Adburl=postgresql://bmiller:@localhost/runestone -Abasecourse=thinkcspy -Adownloads_enabled=false -Adefault_ac_lang=python -Arunestone_version=3.0.7 -Abuild_info=4.0.1 ././_sources/ ./build/thinkcspy
```

#### Error logs / results
```
error: cannot find files [path('./build/thinkcspy')]
```
- <public link of unexpected result if you have>

#### Expected results
build the document

### Reproducible project / your project
https://github.com/RunestoneInteractive/thinkcspy

### Environment info
- OS: Linux and Mac
- Python version: 2.7 and 3.6
- Sphinx version: 1.7.0

**Comment by tk0miya** [comment]
With following Dockerfile, I don't see the such error.
```
FROM tk0miya/sphinx-html

RUN git clone https://github.com/RunestoneInteractive/thinkcspy
WORKDIR /docs/thinkcspy
RUN pip3 install -r requirements.txt
RUN pip3 install Sphinx==1.7.0
RUN sphinx-build  -b html -d ./build/thinkcspy/doctrees -c . -a -E -Acourse_id=thinkcspy -Alogin_required=false -Aappname=runestone -Aloglevel=10 -Acourse_url=http://127.0.0.1:8000 -Ause_services=true -Apython3=true -Adburl=postgresql://bmiller:@localhost/runestone -Abasecourse=thinkcspy -Adownloads_enabled=false -Adefault_ac_lang=python -Arunestone_version=3.0.7 -Abuild_info=4.0.1 ././_sources/ ./build/thinkcspy
```
console log (tail)
```
...
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 2 warnings.

The HTML pages are in build/thinkcspy.
```

Could you check your trouble again?

**Comment by bnmnetp** [comment]
Yes, you are right.

I should have tried to run the sphinx-build command manually before I reported this problem.   Running sphinx-build manually does work.  

However, I have a wrapper program that runs sphinx-build through paver.  which works perfectly with 1.6.3 but fails with the message I reported for 1.7.0.  So there must be some new incompatibility with paver or sphinxcontrib-paverutils that was introduced in 1.7.0

Do you have any idea on where I would start looking to track this down?

**Comment by bnmnetp** [comment]
OK, I'm pretty sure this is due to #3668 (The arguments has changed of main functions for each command) and its impact on sphinx-contrib/paverutils

**Comment by tk0miya** [comment]
OMG. Thank you for letting me know. I try to fix it in #4624. I think it would fix this problem.

**Comment by bnmnetp** [comment]
OK, I'm a little confused -- #3668 was listed as an incompatible change in the release notes.  That change broke paverutils.  Paverutils had been successfully working around the "bug" fixed by #3668 for years...   After some investigation I can see that it is a very small change to paverutils.  

But, it also looks like you are undoing some of what #3668 "fixed". I'm happy either way, but I don't want to do a PR to paverutils that I'll need to undo when 1.7.1 comes out.  OR, am I misunderstanding and there are actually two different problems here??

**Comment by tk0miya** [comment]
At that time, I didn't think `main()` functions are used by third parties. So I modified its interface at #3668 and didn't list it to incompatible changes. Sorry for my lack of imagine.

On the other hand, we decide to move the `sphinx.main()` functions to `sphinx.cmd.build` package since 1.7. As a result, `sphinx.main()` is marked as deprecated, and it will be removed in Sphinx-2.0.

In #4624, I'd like to revert the change of the interface of `sphinx.main()` for old third parties. It works simple wrapper of `sphinx.cmd.build.main()` which has new interface (with lesser argument). After the change, we'll have two main functions:
* `sphinx.main()`: old interface, marked as deprecated (broken only in 1.7.0)
* `sphinx.cmd.build.main()`: new interface, stable

So please call `sphinx.cmd.build.main()` instead for Sphinx >= 1.7. Of course, it's okay to call `sphinx.main()` for a while. But it will be removed in 2.0.

**Comment by tk0miya** [comment]
Fixed #4624. Thank you for reporting!
